stl/inc/xstring

```diff
@@ -2689,19 +2684,8 @@ private:
         auto&& _Alproxy = _GET_PROXY_ALLOCATOR(_Alty, _Al);
         _Container_proxy_ptr<_Alty> _Proxy(_Alproxy, _My_data);
 
+        if (_Count < _BUF_SIZE) {
 #if _HAS_CXX20
-        if (_STD is_constant_evaluated()) {
-            _My_data._Myres = _BUF_SIZE; // TRANSITION: constexpr SSO
-        }
-
-        const bool _Stay_small = _Count < _BUF_SIZE && !_STD is_constant_evaluated();
-#else // ^^^ _HAS_CXX20 / !_HAS_CXX20 vvv
-        const bool _Stay_small = _Count < _BUF_SIZE;
-#endif // _HAS_CXX20
-
-        if (_Stay_small) {
-#if _HAS_CXX20
-            // TRANSITION: This is currently unused until SSO support is merged
             if (_STD is_constant_evaluated()) {
                 _Construct_in_place(_My_data._Bx);

```

Comment:
To improve the debugging experience I would have kept the `is_constant_evaluated` outside the function and just replaced the `_Construct_in_place` with the new helper function that wraps the loop

Reply:
I think it's simpler to have just one line to step over instead of the entire branch structure, but I think we're arguing personal preference. It's your PR, feel free to change things as you like =)

Reply:
I also think the current state of the code is simpler.